### ISSUE TITLE
Draggers are disabled too quickly

### ISSUE THREAD

**Issue by def-** [ddnet/ddnet#5263]
Reported by bencie on Fall into the Future.
Bisect revealed: https://github.com/ddnet/ddnet/pull/5245
```
62c99515d564181e295a1528dc3dc23fbe3e2887 is the first bad commit
commit 62c99515d564181e295a1528dc3dc23fbe3e2887
Author: c0d3d3v <c0d3d3v@mag-keinen-spam.de>
Date:   Thu May 26 19:33:01 2022 +0200

    disable dragger beams soon, not only every 150ms

 src/game/server/entities/dragger.cpp      |  2 +-
 src/game/server/entities/dragger_beam.cpp | 15 +++++++++++++--
 src/game/server/entities/dragger_beam.h   |  2 +-
 3 files changed, 15 insertions(+), 4 deletions(-)
```
Previously: https://youtu.be/a7_4szWQ2Pc?t=32 (0:32)
Currently: 
https://user-images.githubusercontent.com/2335377/170823696-9e8ee776-8d00-4287-85a0-d48c77d6a39c.mp4
So it seems like the draggers are now being disabled too quickly instead of too slowly. Would be great to check how it was before and get back the exact same physics.

**Comment by def-** [comment]
The map is now unfinishable. I can't just revert the change since then the map is too easy. I don't want to revert all of the dragger rework (~20 changes)

**Comment by C0D3D3V** [comment]
mh strange. I will investigate. Maybe it is not the timing... But I can reproduce it, and I compared it with 15.9.1 so yes seems broken currently

**Comment by C0D3D3V** [comment]
So this is the log of the first part:
For 15.9.1:
```
[2022-05-28 15:46:16][dragger]: drag - tick=539 | vel=(0.666523, -0.745484) | player_pos (1649.000000, 1649.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=540 | vel=(0.998555, -0.991578) | player_pos (1649.000000, 1649.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=541 | vel=(1.166523, -1.237672) | player_pos (1649.000000, 1649.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=542 | vel=(1.248417, -1.483888) | player_pos (1650.000000, 1648.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=543 | vel=(1.291248, -1.730105) | player_pos (1651.000000, 1647.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=544 | vel=(1.310640, -1.976323) | player_pos (1652.000000, 1646.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=545 | vel=(1.322219, -2.222542) | player_pos (1653.000000, 1645.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=546 | vel=(1.327158, -2.467712) | player_pos (1654.000000, 1643.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=547 | vel=(1.929764, -2.712871) | player_pos (1655.000000, 1641.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=548 | vel=(2.499805, -2.955302) | player_pos (1657.000000, 1639.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=549 | vel=(3.042499, -3.201642) | player_pos (1659.000000, 1637.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=550 | vel=(3.557704, -3.448112) | player_pos (1662.000000, 1634.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=551 | vel=(4.045557, -3.694589) | player_pos (1665.000000, 1631.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=552 | vel=(4.508577, -3.942309) | player_pos (1669.000000, 1628.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:16][dragger]: drag - tick=553 | vel=(4.944204, -4.190066) | player_pos (1673.000000, 1625.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:17][dragger]: drag - tick=554 | vel=(5.356172, -4.441916) | player_pos (1678.000000, 1621.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:17][dragger]: drag - tick=555 | vel=(5.748533, -4.693821) | player_pos (1683.000000, 1617.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:17][dragger]: drag - tick=556 | vel=(6.117378, -4.949690) | player_pos (1688.000000, 1613.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:17][dragger]: drag - tick=557 | vel=(6.465066, -5.206951) | player_pos (1694.000000, 1609.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:17][dragger]: drag - tick=558 | vel=(6.790509, -5.467065) | player_pos (1700.000000, 1604.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:17][dragger]: drag - tick=559 | vel=(7.096295, -5.731175) | player_pos (1706.000000, 1599.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:17][dragger]: drag release switch target - tick=560 | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:17][dragger]: drag - tick=602 | vel=(0.995314, 14.793575) | player_pos (1825.000000, 1825.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:17][dragger]: drag - tick=603 | vel=(1.155363, 14.315278) | player_pos (1826.000000, 1840.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=604 | vel=(1.300200, 13.837133) | player_pos (1827.000000, 1855.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=605 | vel=(1.430143, 13.355288) | player_pos (1828.000000, 1869.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=606 | vel=(1.548682, 12.873550) | player_pos (1829.000000, 1883.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=607 | vel=(1.656109, 12.391972) | player_pos (1830.000000, 1896.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=608 | vel=(1.750420, 11.906271) | player_pos (1832.000000, 1909.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=609 | vel=(1.833580, 11.420723) | player_pos (1834.000000, 1921.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=610 | vel=(1.905264, 10.935262) | player_pos (1836.000000, 1933.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=611 | vel=(1.965709, 10.446013) | player_pos (1838.000000, 1944.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=612 | vel=(2.018534, 9.956832) | player_pos (1840.000000, 1955.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=613 | vel=(2.063955, 9.467745) | player_pos (1842.000000, 1965.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=614 | vel=(2.101716, 8.978709) | player_pos (1844.000000, 1975.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=615 | vel=(2.132016, 8.489749) | player_pos (1846.000000, 1984.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=616 | vel=(2.158531, 7.996923) | player_pos (1848.000000, 1993.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=617 | vel=(2.177536, 7.504160) | player_pos (1850.000000, 2001.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=618 | vel=(2.192733, 7.011428) | player_pos (1852.000000, 2009.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag - tick=619 | vel=(2.200210, 6.518726) | player_pos (1854.000000, 2017.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag release Res - tick=620 | vel=(2.089844, 7.019531) | player_pos (1856.000000, 2024.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:46:18][dragger]: drag release Res - tick=623 | vel=(1.792969, 8.519531) | player_pos (1862.000000, 2049.000000) | dragger_pos (1936.000000, 1328.000000)
```

For latest master:
```
[2022-05-28 15:45:20][dragger]: start - tick=686 | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:20][beam]: drag - tick=687 | vel=(0.666523, -0.745484) | player_pos (1649.000000, 1649.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:20][beam]: drag - tick=688 | vel=(0.998555, -0.991578) | player_pos (1649.000000, 1649.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:20][beam]: drag - tick=689 | vel=(1.166523, -1.237672) | player_pos (1649.000000, 1649.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=690 | vel=(1.248417, -1.483888) | player_pos (1650.000000, 1648.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=691 | vel=(1.291248, -1.730105) | player_pos (1651.000000, 1647.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=692 | vel=(1.310640, -1.976323) | player_pos (1652.000000, 1646.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=693 | vel=(1.322219, -2.222542) | player_pos (1653.000000, 1645.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=694 | vel=(1.327158, -2.467712) | player_pos (1654.000000, 1643.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=695 | vel=(1.929764, -2.712871) | player_pos (1655.000000, 1641.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=696 | vel=(2.499805, -2.955302) | player_pos (1657.000000, 1639.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=697 | vel=(3.042499, -3.201642) | player_pos (1659.000000, 1637.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=698 | vel=(3.557704, -3.448112) | player_pos (1662.000000, 1634.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=699 | vel=(4.045557, -3.694589) | player_pos (1665.000000, 1631.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=700 | vel=(4.508577, -3.942309) | player_pos (1669.000000, 1628.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=701 | vel=(4.944204, -4.190066) | player_pos (1673.000000, 1625.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=702 | vel=(5.356172, -4.441916) | player_pos (1678.000000, 1621.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=703 | vel=(5.748533, -4.693821) | player_pos (1683.000000, 1617.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=704 | vel=(6.117378, -4.949690) | player_pos (1688.000000, 1613.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=705 | vel=(6.465066, -5.206951) | player_pos (1694.000000, 1609.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=706 | vel=(6.790509, -5.467065) | player_pos (1700.000000, 1604.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag - tick=707 | vel=(7.096295, -5.731175) | player_pos (1706.000000, 1599.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:21][beam]: drag release switch - tick=708 | vel=(6.742188, -5.230469) | player_pos (1713.000000, 1594.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][dragger]: start - tick=749 | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=750 | vel=(0.995314, 14.793575) | player_pos (1825.000000, 1825.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=751 | vel=(1.155363, 14.315278) | player_pos (1826.000000, 1840.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=752 | vel=(1.300200, 13.837133) | player_pos (1827.000000, 1855.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=753 | vel=(1.430143, 13.355288) | player_pos (1828.000000, 1869.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=754 | vel=(1.548682, 12.873550) | player_pos (1829.000000, 1883.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=755 | vel=(1.656109, 12.391972) | player_pos (1830.000000, 1896.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=756 | vel=(1.750420, 11.906271) | player_pos (1832.000000, 1909.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=757 | vel=(1.833580, 11.420723) | player_pos (1834.000000, 1921.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=758 | vel=(1.905264, 10.935262) | player_pos (1836.000000, 1933.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=759 | vel=(1.965709, 10.446013) | player_pos (1838.000000, 1944.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=760 | vel=(2.018534, 9.956832) | player_pos (1840.000000, 1955.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=761 | vel=(2.063955, 9.467745) | player_pos (1842.000000, 1965.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=762 | vel=(2.101716, 8.978709) | player_pos (1844.000000, 1975.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=763 | vel=(2.132016, 8.489749) | player_pos (1846.000000, 1984.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=764 | vel=(2.158531, 7.996923) | player_pos (1848.000000, 1993.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=765 | vel=(2.177536, 7.504160) | player_pos (1850.000000, 2001.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=766 | vel=(2.192733, 7.011428) | player_pos (1852.000000, 2009.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag - tick=767 | vel=(2.200210, 6.518726) | player_pos (1854.000000, 2017.000000) | dragger_pos (1936.000000, 1328.000000)
[2022-05-28 15:45:22][beam]: drag release Res - tick=768 | vel=(2.089844, 7.019531) | player_pos (1856.000000, 2024.000000) | dragger_pos (1936.000000, 1328.000000)
```

looks the same, thats why I thought, the dragger beam releases every tick (but actually as I now looked again in the old code it does not look like it releases every tick). I will now check the part from the video

**Comment by C0D3D3V** [comment]
Ok I know now the problem.

When I did the last fix (https://github.com/ddnet/ddnet/pull/5245) I probably messed up the diffs somehow and thought the draggers were really disabled every tick. It looked perfectly identical (even the logs from above look almost identical). 
But in fact they are not identical at all. In the old code, the checks were performed every 7th server tick (in the comment it says every 150ms, actually it's more like 140ms), I had taken over so. Only I have introduced the Dragger Beams. However, these are created as "new" entities, and thus end up in the tick order at the very front, whereby they are not executed in the same tick as they were created (this can also be seen in the log). And the first part is a tick perfect part, so because I was one tick off, the part did not work anymore. however, with the fix it worked again because by chance it  turned the draggers off exactly in the right tick.

Long story short. I'm about to make a fix PR.
